src/main/python/pgshovel/replication/streams/kafka.py

```diff
@@ -40,12 +54,44 @@ def consume(self, state):
             consumer.seek(delta, 1)
             assert consumer.offsets[0] == offset
 
-        for offset, message in consumer:
-            decoded = self.codec.decode(message.value)
-            state = validate_state(state, offset, decoded)
+        for offset, message in self.__primed(consumer, state):
+            state = validate_state(state, offset, message)
             # XXX: This is necessary because of a bug in protocol buffer oneof.
             state = type(state).FromString(state.SerializeToString())
-            yield state, offset, decoded
+            yield state, offset, message
+
+    def __primed(self, consumer, state, attempts=5):
```

Comment:
> Don't want to prematurely optimize for an abstract case that isn't necessary at right this moment, vs having some more experience with code and doing it later.

Trying to avoid going off on a rant here, but I fundamentally disagree with categorizing this as a "premature optimization."